src/app/_components/SmartImage.tsx

```diff
@@ -0,0 +1,86 @@
+'use server'
+
+import fs from 'fs/promises'
+import path from 'path'
+
+import { type ImageProps } from 'next/image'
+import Image from 'next/image'
+
+import type { ImageObjectFit, StaticImageProps } from '@/types/imageType'
+
+import { graphicsData } from '@/data/graphicsData'
+
+const { data: fallbackSrc, alt: fallbackAlt } = graphicsData.imageFallback
+
+export type SmartImageProps = {
+  src?: string | StaticImageProps['data']
+  alt?: string
+  className?: string
+  objectFit?: ImageObjectFit
+  padding?: boolean
+} & Omit<ImageProps, 'src' | 'alt' | 'className'>
+
+export async function SmartImage({
```

Comment:
What do you think about something like this?

Does it read a bit better to you?


```ts
export async function SmartImage({ src, alt = '', ...props }: SmartImageProps) {
  if (!src) {
    return <Image src={fallbackSrc} alt={fallbackAlt} {...props} />
  }

  const isValid = await checkImageValidity(src)

  return (
    <Image
      src={isValid ? src : fallbackSrc}
      alt={isValid ? alt : fallbackAlt}
      {...props}
    />
  )
}

async function checkImageValidity(src: NonNullable<SmartImageProps['src']>) {
  const isStaticImport = typeof src === 'object'
  const isAssetImage = typeof src === 'string' && src.startsWith('/assets')

  if (isStaticImport) {
    return true
  }

  if (isAssetImage) {
    return await checkAssetImageExists(src)
  }

  return await checkRemoteImageExists(src)
}
```

Reply:
I see what you mean, personally I kind of like the separation of logic (`getImageSrc`) and ui rendering (`SmartImage`). That said, I’m happy to explore versions if you think this doesn’t work well for the current context.